### DISCUSSION TITLE
Unknown ParameterID in captured traffic

### DISCUSSION THREAD

**Discussion by mechaniputer** [eProsima/Fast-DDS#5239] [Q&A] [answered]
Hello,

I am capturing FastDDS traffic with Wireshark and trying to understand the packets. In the initial announcement packet, I am seeing a field that I cannot identify with Wireshark or by reading the RTPS specification.

In the DATA (0x15) submessage, in the serializedData portion with PL_CDR_LE encapsulation, there is a field that Wireshark says is "Unknown". It is located after PID_PARTICIPANT_GUID but before PID_METATRAFFIC_UNICAST_LOCATOR.

The ParameterID value is 0x8007, the length is 4, and the data is 11000000. What is the possible significance of this field?

Thanks in advance.

**Comment by MiguelCompany** [comment, marked as answer]
When a parameter ID has the most significant bit set (i.e. it is >= 0x8000), it means it is a custom (non-standard) parameter that should not be interpreted by vendors different from the one originating the message.

That is why Wireshark is not showing the interpretation of that ID.

The list of custom parameter IDs for Fast DDS is [here](https://github.com/eProsima/Fast-DDS/blob/5abe0c0e60639ebf524d916b774025b3fdaae661/include/fastdds/dds/core/policy/ParameterTypes.hpp#L168-L174).

**Reply by mechaniputer** [reply]
Thanks. Looks like it is: PID_NETWORK_CONFIGURATION_SET
Still not sure what it actually does, but I found the code that ignores it when the vendor is not the same as the local one.
